app/services/ipfs.js

```diff
@@ -26,6 +26,9 @@ export default Ember.Service.extend({
   getFile(hash) {
     return this.get('ipfs').cat(hash, { buffer: true }).then(res => {
       return res.toString();
+    }, err => {
+      Ember.Logger.error('[ipfs] error trying to fetch file', hash, err);
+      return err;
```

Comment:
I think you want to `throw` the error here. Otherwise the `getFile` Promise will resolve with the `err` as value instead of being rejected.